### PR TITLE
Tweak: Replace jQuery code with vanilla JS in frontend [ED-13040]

### PR DESCRIPTION
**Pull request by rami-elementor**

### CODE REVIEW

File: assets/dev/js/frontend/hello-frontend.js [hunk 1]
```diff
@@ -17,62 +17,65 @@ class elementorHelloThemeHandler {
...
                 if ( ENTER_KEY === event.keyCode || SPACE_KEY === event.keyCode ) {
                     event.currentTarget.click();
                 }
             } );
-        this.elements.$dropdownMenu.on( 'click', '.menu-item-has-children > a', this.handleMenuChildren );
+        this.elements.dropdownMenu.addEventListener( 'click', ( event ) => {
+            if ( event.target.closest( '.menu-item-has-children > a' ) ) {
+                this.handleMenuChildren.bind( this );
+            }
+        } );
```
[the remark this answers is not in the document]
Comment (bainternet) [reply assets/dev/js/frontend/hello-frontend.js:41]: its not the same as before,

the original targeted the `a` directly, and here you are "guessing" the intent.

it should be something like:

```Javascript
const menuChildren = this.elements.dropdownMenu.querySelectorAll( '.menu-item-has-children > a' );
[ ...menuChildren ].forEach( ( anchorElement ) => anchorElement.addEventListener( 'click', ( event ) => this.handleMenuChildren( event ) );

**Comment by bainternet** [comment]
Also why remove all Arrow functions?

**Comment by ramiy** [comment]
> Also why remove all Arrow functions?

I tested the code, it didn't worked with arrow functions. Scoping issues I guess.

**Review by bainternet — APPROVED** [review]

**Review by hein-obox — APPROVED** [review]

**Review by louiswol94 — APPROVED** [review]
@rami-elementor You might want to consider adding optional chaining on all the `classList` properties. jQuery is a bit more forgiving in this regard, but plain JS will throw an error.

**Comment by rami-elementor** [comment]
> @rami-elementor You might want to consider adding optional chaining on all the `classList` properties. jQuery is a bit more forgiving in this regard, but plain JS will throw an error.

Good point. Thank you @louiswol94.

**Review by bainternet — APPROVED** [review]

**Pull request merged**